application/src/command.rs

```diff
@@ -0,0 +1,55 @@
+use crate::port::{TodoListRepository, TodoListStore};
+use domain::todolist_message::TodoListMessage;
+use framework::*;
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+pub enum Command {
+    AddTask { name: String },
+    RemoveTask { index: usize },
+    CompleteTask { index: usize },
+}
+
+#[derive(Debug, Error)]
+pub enum CommandError {
+    #[error("Task name cannot be empty")]
+    TaskNameCannotBeEmpty,
+}
```

Comment:
**Suggestion:** Consider adding a docstring to the `CommandError` enum. [best practice]
```suggestion
#[derive(Debug, Error)]
/// Represents an error that can occur while executing a command on the todo list.
pub enum CommandError {
```